### PR TITLE
Initial commit for #75 - setting up master ELB

### PR DESCRIPTION
**Pull request by tayzlor**

### CODE REVIEW

**Comment by tayzlor** [comment]
I can't get the health check for the ELB to come back (on HTTP port). I must be missing a config option somewhere to allow it (some debug output from inside an instance below):

```
ip-10-0-1-111 ~ # curl -v 10.0.1.111:8080/healthz
*   Trying 10.0.1.111...
* connect to 10.0.1.111 port 8080 failed: Connection refused
* Failed to connect to 10.0.1.111 port 8080: Connection refused
* Closing connection 0
curl: (7) Failed to connect to 10.0.1.111 port 8080: Connection refused
ip-10-0-1-111 ~ # curl -v 127.0.0.1:8080/healthz
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /healthz HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.47.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Tue, 26 Apr 2016 17:17:58 GMT
< Content-Length: 2
< Content-Type: text/plain; charset=utf-8
<
* Connection #0 to host 127.0.0.1 left intact
```

File: terraform/aws/elb/main.tf [hunk 1]
```diff
@@ -29,17 +19,33 @@ resource "aws_elb" "elb" {
     interval            = 30
   }
 
+  listener {
+    instance_port     = 443
+    instance_protocol = "tcp"
+    lb_port           = 443
+    lb_protocol       = "tcp"
+  }
+
```
Comment (wallies) [inline terraform/aws/elb/main.tf]: this needs to be more like this 
```
listener {
    instance_port = 443
    instance_protocol = "tcp"
    lb_port = 443
    lb_protocol = "https"
    ssl_certificate_id = "${var.ssl_certificate}"
}
```

Comment (tayzlor) [inline terraform/aws/elb/main.tf]: how do u propose to get the certificate in there though? is it actually needed? its only needed if you want to terminate SSL at the ELB - my current config just passes it through

**Comment by stevenjack** [comment]
@tayzlor You need to also add `--insecure-bind-address=0.0.0.0` as one of the `api-server` options, as currently it's only bound to localhost.

**Comment by tayzlor** [comment]
https://github.com/Capgemini/kubeform/pull/77/commits/379406fbb9b34ba54805ff4e862e450703e1c9c3 moves this on a bit further - ELB now set up and forwarding traffic on to the API servers. We now have an issue with the certificate -

```
E0505 14:42:00.147135       1 event.go:202] Unable to write event: 'Post https://kube-master-1693761008.eu-west-1.elb.amazonaws.com:443/api/v1/namespaces/default/events: x509: failed to load system roots and no roots provided' (may retry after sleeping)
```

**Comment by tayzlor** [comment]
Making a note here that we probably need to generate and pass in a kubeconfig file for the kube-proxy ```--kubeconfig=/var/lib/kubelet/kubeconfig```

**Comment by tayzlor** [comment]
https://github.com/Capgemini/kubeform/pull/77/commits/a0847dee7adbc628f47a8ddf5a02d50b7b5681b6 fixes HTTPS connections. It doesn't actually fix #79 but we could fix that in a follow-up.

**Pull request merged**